components/shared/profile/ProfileHeader.tsx

```diff
@@ -1,29 +1,74 @@
 import React from "react";
 import Image from "next/image";
-import { getSession } from "@/lib/authOptions";
 
-const ProfileHeader = async () => {
-  const session = await getSession();
+import { FiMapPin } from "react-icons/fi";
+import { FaLink } from "react-icons/fa6";
+import { CiCalendar } from "react-icons/ci";
 
+import { redirect } from "next/navigation";
+import { IUser } from "@/database/models/user.model";
+import { format } from "date-fns";
+import { Button } from "@/components/ui/button";
+import Link from "next/link";
+import icnEdit from "@/public/icons/icn-edit.svg";
+interface ProfileHeaderProps {
+  user: Partial<IUser>;
+}
+const ProfileHeader = async ({ user }: ProfileHeaderProps) => {
+  if (!user) {
+    redirect("/sign-in");
+  }
   return (
-    <div className={`mt-[40px] flex h-[90px] items-center `}>
-      <div className={` flex-0 ml-[10px]`}>
+    <section
+      className={`mt-[40px] flex h-[90px] items-center justify-between gap-5`}
+    >
+      <div className={` flex-0 justify-start gap-5`}>
```

Comment:
```suggestion
      <div className="flex-0 justify-start gap-5 shrink-0">
```
Check mobile responsiveness for stuff like this. Adding shrink-0 will tell this div with the image to not resize.

We also don't need the brackets or backticks.